src/main/java/com/aws/greengrass/deployment/DeviceConfiguration.java

```diff
@@ -227,7 +201,8 @@ public String getNucleusComponentName() {
         try (LockScope ls = LockScope.lock(lock)) {
```

Comment:
this whole method does not belong here I think. Try and move it out

Reply:
Tried moving out, encountered some dependency issues.
DeviceConfiguration needs to initialize nucleus component name and config because it's used to search up topic values in the constructor.

It doesn't look like a simple task to do